### PR TITLE
Add Kubernetes-SD metrics

### PR DESCRIPTION
**Pull request by dominikschulz**
This PR adds metrics for the new Kubernetes SD implementation.

I'm not quite sure about the metric and label naming. Please advise.

### CODE REVIEW

File: retrieval/discovery/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -35,8 +36,40 @@ const (
...
+)
+
+var (
+	eventCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "sd_kubernetes_events_total",
+			Help:      "The number of Kubernetes events received.",
+		},
+		[]string{"type", "event"},
```
Comment (brian-brazil) [inline retrieval/discovery/kubernetes/kubernetes.go]: The label name "type" should be avoided, as it's very generic. "role" would be a good choice here.

**Comment by brancz** [comment]
Technically there are also relevant events [here](https://github.com/dominikschulz/prometheus/blob/e1e30f12cd5b116abe0d0165e573bee1a924cad2/retrieval/discovery/kubernetes/endpoints.go#L114-L116). The question is whether this is the correct approach, as the config can be duplicated, then the metric will be increased because there are multiple event handlers and each of them increases the metric, however the event is actually only received once. Wdyt @fabxc ?

**Comment by brancz** [comment]
Okay, makes more sense when we treat it as events handled 🙂 . Then there are still events being handled [here](https://github.com/dominikschulz/prometheus/blob/e1e30f12cd5b116abe0d0165e573bee1a924cad2/retrieval/discovery/kubernetes/endpoints.go#L114-L116) though.

**Comment by brancz** [comment]
Thanks for baring with me :) lgtm if @fabxc doesn't have any other comments

**Comment by fabxc** [comment]
Sorry about not responding sooner.

:+1: In general.

This needs to be rebased. "endpoint" should be "endpoints" to be aligned with the role naming.
I'll rebase and fix and merge directly into master.

**Pull request merged**